src/OrchardCore/OrchardCore.DisplayManagement/ShapeFactoryExtensions.cs

```diff
@@ -0,0 +1,187 @@
+        where TModel : class
+    {
+        return factory.CreateAsync<TModel>(shapeType, initializeAsync: (model) =>
+        {
+            initialize?.Invoke(model);
+
+            return ValueTask.CompletedTask;
+        });
+    }
+
+    /// <summary>
+    /// Creates a generic shape and initializes it with some properties.
+    /// </summary>
+    public static ValueTask<IShape> CreateAsync(this IShapeFactory factory, string shapeType, INamedEnumerable<object> parameters)
+    {
+        if (parameters == null || parameters == Arguments.Empty)
+        {
+            return factory.CreateAsync(shapeType);
+        }
+
+        return factory.CreateAsync(shapeType, _newShape, null, createdContext =>
+        {
+            var shape = (Shape)createdContext.Shape;
+
+            // If only one non-Type, use it as the source object to copy.
+            var initializer = parameters.Positional.SingleOrDefault();
+
+            if (initializer != null)
+            {
+                // Use the Arguments class to optimize reflection code.
+                var arguments = Arguments.From(initializer);
+
+                foreach (var prop in arguments.Named)
+                {
+                    shape.Properties[prop.Key] = prop.Value;
+                }
+            }
+            else
+            {
+                foreach (var kv in parameters.Named)
+                {
+                    shape.Properties[kv.Key] = kv.Value;
+                }
+            }
+        });
+    }
+
+    /// <summary>
+    /// Creates a dynamic proxy instance for the type and initializes it.
+    /// </summary>
+    /// <typeparam name="TModel">The type to instantiate.</typeparam>
+    /// <param name="factory">The <see cref="IShapeFactory"/>.</param>
+    /// <param name="initializeAsync">The initialization method.</param>
+    /// <returns></returns>
+    public static ValueTask<IShape> CreateAsync<TModel>(this IShapeFactory factory, Func<TModel, ValueTask> initializeAsync)
+        where TModel : class
+        => factory.CreateAsync(typeof(TModel).Name, initializeAsync);
+
+    /// <summary>
+    /// Creates a dynamic proxy instance for the type and initializes it.
+    /// </summary>
+    /// <typeparam name="TModel">The type to instantiate.</typeparam>
+    /// <param name="factory">The <see cref="IShapeFactory"/>.</param>
+    /// <param name="shapeType">The shape type to create.</param>
+    /// <param name="initializeAsync">The initialization method.</param>
+    /// <returns></returns>
+    public static ValueTask<IShape> CreateAsync<TModel>(this IShapeFactory factory, string shapeType, Func<TModel, ValueTask> initializeAsync)
+        where TModel : class
+    {
+        return factory.CreateAsync(shapeType, () => ShapeFactory(initializeAsync));
+
+        static ValueTask<IShape> ShapeFactory(Func<TModel, ValueTask> init)
```

Comment:
Why you are using internal function it's used once!!

Reply:
@hishamco this code was just moved. Nothing changes to it. and I don't want to make changes to existing code.

Reply:
I see